### PR TITLE
feat: added support for errorText

### PR DESCRIPTION
**Pull request by hanne-at-atb**
As part of https://github.com/AtB-AS/kundevendt/issues/4275, we need to support error state for TextInputSectionItems.

Implemented to look like this:
image

### CODE REVIEW

**Review by marius-at-atb — APPROVED** [review]
Lgtm! :)

File: src/components/sections/items/TextInputSectionItem.tsx [hunk 1]
```diff
@@ -102,40 +115,56 @@ export const TextInputSectionItem = forwardRef<InternalTextInput, TextProps>(
...
+        {errorText !== undefined && (
+          <View
+            style={styles.error}
+            accessibilityLiveRegion={'polite'}
+            accessibilityLabel={`${t(
+              dictionary.messageTypes.error,
+            )}, ${errorText}`}
+          >
+            <ThemeIcon svg={Error} />
+            <ThemeText type={'body__secondary'} style={styles.errorMessage}>
```
[the remark this answers is not in the document]
Comment (hanne-at-atb) [reply src/components/sections/items/TextInputSectionItem.tsx]: Fixed both places :)

File: src/components/sections/items/TextInputSectionItem.tsx [hunk 2]
```diff
@@ -102,40 +115,56 @@ export const TextInputSectionItem = forwardRef<InternalTextInput, TextProps>(
...
+              >
+                <ThemeIcon svg={Close} />
+              </PressableOpacity>
+            </View>
+          ) : null}
+        </View>
+        {errorText !== undefined && (
+          <View
+            style={styles.error}
+            accessibilityLiveRegion="polite"
```
Comment (gorandalum) [inline src/components/sections/items/TextInputSectionItem.tsx:157]: Nice solution! I took a look at some aria doc to see if a specific role also would be appropriate, and think maybe role "alert" would be nice here. Some cliffs from the doc:
> The alert role is used to communicate an important and usually time-sensitive message to the user. When this role is added to an element, the browser will send out an accessible alert event to assistive technology products which can then notify the user.
>
> The alert role should only be used for information that requires the user's immediate attention, for example:
>
>- An invalid value was entered into a form field

Maybe check if when adding accessibility role `alert`, is the announcements done even without adding live region?

**Review by gorandalum — APPROVED** [review]
Very good!

File: src/components/sections/items/TextInputSectionItem.tsx (hunk 2 above)
Comment (hanne-at-atb) [reply src/components/sections/items/TextInputSectionItem.tsx]: Added `role: 'alert'` now. Does however look like accessibilityLiveRegion is only supported by Android, so exchanged it with focusRef that in turn will trigger the screen reader to read out error message when it appears for both android and ios.

File: src/components/sections/items/TextInputSectionItem.tsx [hunk 3]
```diff
@@ -183,6 +184,7 @@ const useInputStyle = StyleSheet.createTheme((theme) => ({
     borderColor: theme.static.background.background_0.background,
   },
   containerInline: {
+    backgroundColor: 'red',
```
Comment (marius-at-atb) [inline src/components/sections/items/TextInputSectionItem.tsx:187]: Remember to remove this one :)

**Pull request merged**